esp-hal/src/ledc/mod.rs

```diff
@@ -51,6 +53,19 @@
 //!         pin_config: channel::config::PinConfig::PushPull,
 //!     })
 //!     .unwrap();
+//!
+//! channel0.start_duty_fade(0, 100, 2000).expect_err(
+//!     "Fading from 0% to 100%, at 24kHz and 5-bit resolution, over 2 seconds,
+//! should fail", );
```

Comment:
rustfmt doesn't always give us a nice doctest by itself. Could you break this up into lines so the string isn't wrapped?

Reply:
Or, just shorten the string, I guess

Reply:
Also should we teach things that fail?

Reply:
LEDC is due for a complete overhaul, I'm not going to lose sleep over these docs so lets just merge as is

Reply:
but in general, I agree that we shouldn't teach failure modes in docs, it's not very helpful